moveit_ros/planning/moveit_cpp/include/moveit/moveit_cpp/planning_component.h

```diff
@@ -208,6 +209,11 @@ class PlanningComponent
   const planning_interface::MotionPlanResponse& getLastMotionPlanResponse();
 
 private:
+  /** \brief A callback function that is used as the default parallel planning stop criterion.
+   *          It stops parallel planning as soon as any planner finds a solution. */
+  static bool stopAtFirstSolution(PlanSolutions const& plan_solutions,
+                                  PlanningComponent::MultiPipelinePlanRequestParameters const& plan_request_parameters);
```

Comment:
I tried to make this a free function, but:

It needs knowledge of `PlanningComponent::MultiPipelinePlanRequestParameters`. [Apparently you can't forward-declare a struct nested in a class in C++](https://stackoverflow.com/questions/951234/forward-declaration-of-nested-types-classes-in-c), so... I don't think a free function is feasible.

Reply:
If we don't make it a default, you could make it a free function. Another option would be to move `MultiPipelinePlanRequestParameters` out of the `PlanningComponentClass`. The main reason to keep it there was to treat it similarly as `PlanRequestParameters` and the lack of a need to keep it outside of the class but that would justify moving it out

Reply:
This is tricky. `MultiPipelinePlanRequestParameters` needs knowledge of `PlanRequestParameters` at L112:

`std::vector<PlanRequestParameters> multi_plan_request_parameters;`

Reply:
I moved the new callback to a new translation unit. Seems ok to me.